models.py

```diff
@@ -559,11 +560,26 @@ def _urls_and_domains(self, auth_entity, user_url):
     for i, url in enumerate(candidates):
       final, domain, ok = util.get_webmention_target(url, resolve=i < MAX_AUTHOR_URLS)
       if ok:
-        if util.schemeless(final.lower()).startswith(util.schemeless(url.lower())):
+        current_url = ''
+        final = final.lower()
+        # Ignore redirects when fetching the root url, they are dealt with next.
+        if util.schemeless(final) == util.schemeless(url.lower()):
+          # If final has a path segment check if root has a matching rel=me.
+          match = re.match(r'^(https?://[^/]+)/.+', final)
```

Comment:
heh. i started to ask for `urlparse` instead, but then realized you also need to extract just the root URL. doable with `urlparse` too, but annoying. this lgtm. :P